common/src/main/java/com/microsoft/identity/common/internal/telemetry/events/ApiStartEvent.java

```diff
@@ -0,0 +1,198 @@
+        types(TELEMETRY_EVENT_API_EVENT);
+    }
+
+    @Override
+    public ApiStartEvent put(@NonNull final String propertyName, final String propertyValue) {
+        super.put(propertyName, propertyValue);
+        return this;
+    }
+
+    public ApiStartEvent putProperties(OperationParameters parameters) {
+        this.put(TELEMETRY_KEY_AUTHORITY, parameters.getAuthority().toString());
+        this.put(TELEMETRY_KEY_AUTHORITY_TYPE, parameters.getAuthority().getAuthorityTypeString());
+        this.put(TELEMETRY_KEY_SDK_NAME, parameters.getSdkType().name());
+        this.put(TELEMETRY_KEY_SDK_VERSION, parameters.getSdkVersion());
+        this.put(TELEMETRY_KEY_CLAIM_REQUEST, StringUtil.isEmpty(
+                parameters.getClaimsRequestJson())? TELEMETRY_VALUE_FALSE : TELEMETRY_VALUE_TRUE
+        );
+        this.put(TELEMETRY_KEY_REDIRECT_URI, parameters.getRedirectUri());
+        this.put(TELEMETRY_KEY_CLIENT_ID, parameters.getClientId());
+
+        if (parameters instanceof AcquireTokenOperationParameters) {
+            this.put(
+                    TELEMETRY_KEY_USER_AGENT,
+                    ((AcquireTokenOperationParameters) parameters).getAuthorizationAgent().name()
+            );
+
+            this.put(TELEMETRY_KEY_LOGIN_HINT,
+                    ((AcquireTokenOperationParameters) parameters).getLoginHint()
+            );
+
+            if (null != ((AcquireTokenOperationParameters) parameters).getExtraQueryStringParameters()) {
+                this.put(TELEMETRY_KEY_REQUEST_QUERY_PARAMS,
+                        String.valueOf(((AcquireTokenOperationParameters) parameters).getExtraQueryStringParameters().size())
+                );
+            }
+        }
+
+        if (parameters instanceof AcquireTokenSilentOperationParameters) {
+            this.put(TELEMETRY_KEY_USER_ID, parameters.getAccount().getHomeAccountId());
+            this.put(
+                    TELEMETRY_KEY_IS_FORCE_REFRESH,
+                    String.valueOf(((AcquireTokenSilentOperationParameters) parameters).getForceRefresh())
+            );
+            this.put(
+                    TELEMETRY_KEY_BROKER_PROTOCOL_VERSION,
+                    String.valueOf(((AcquireTokenSilentOperationParameters) parameters).getRequiredBrokerProtocolVersion())
+            );
+            this.put(TELEMETRY_KEY_SCOPE_SIZE, String.valueOf(parameters.getScopes().size()));
+            this.put(TELEMETRY_KEY_SCOPE, parameters.getScopes().toString());
+        }
+
+        if (parameters instanceof BrokerAcquireTokenOperationParameters) {
+            //TODO when integrate the telemetry with broker.
+        }
+
+        if (parameters instanceof BrokerAcquireTokenSilentOperationParameters) {
+            //TODO when integrate the telemetry with broker.
+        }
+
+        return this;
+    }
+
+    public ApiStartEvent authority(@NonNull final String authority) {
+        super.put(TELEMETRY_KEY_AUTHORITY, sanitizeUrlForTelemetry(authority));
```

Comment:
This class seems to arbitrarily switch between calling `super.put` and `this.put` - why not

 `return put(TELEMETRY_KEY_AUTHORITY, sanitizeUrlForTelemetry(authority));`